src/marginaldamage.jl

```diff
@@ -1,55 +1,88 @@
 Computes the social cost of CO2 for an emissions pulse in `year` for the provided MimiRICE2010 model. 
 If no model is provided, the default model from MimiRICE2010.get_model() is used.
 Constant discounting is used from the specified pure rate of time preference `prtp`.
 """
-function compute_scc_mm(m::Model=get_model(); year::Union{Int, Nothing} = nothing, last_year::Int = model_years[end], prtp::Float64 = 0.015, eta::Float64=1.5)
+function compute_scc_mm(m::Model=get_model(); year::Union{Int, Nothing} = nothing, 
+    last_year::Int = model_years[end], prtp::Float64 = 0.015, eta::Float64=1.5, 
+    equity_weighting::Bool = false, normalization_region::Union{Int, Nothing} = nothing)
+
     year === nothing ? error("Must specify an emission year. Try `compute_scc_mm(m, year=2015)`.") : nothing
     !(last_year in model_years) ? error("Invlaid value of $last_year for last_year. last_year must be within the model's time index $model_years.") : nothing
     !(year in model_years[1]:10:last_year) ? error("Cannot compute the scc for year $year, year must be within the model's time index $(model_years[1]):10:$last_year.") : nothing
 
     mm = get_marginal_model(m; year = year)
-    scc = _compute_scc(mm; year=year, last_year=last_year, prtp=prtp, eta=eta)
+    scc = _compute_scc(mm; year=year, last_year=last_year, prtp=prtp, eta=eta, equity_weighting=equity_weighting, normalization_region=normalization_region)
     
     return (scc = scc, mm = mm)
 end
 
 # helper function for computing SCC from a MarginalModel, not to be exported or advertised to users
-function _compute_scc(mm::MarginalModel; year::Int, last_year::Int, prtp::Float64, eta::Float64)
+function _compute_scc(mm::MarginalModel; year::Int, last_year::Int, prtp::Float64, eta::Float64, equity_weighting::Bool = false, normalization_region::Union{Int, Nothing} = nothing)
     ntimesteps = findfirst(isequal(last_year), model_years)     # Will run through the timestep of the specified last_year 
     run(mm, ntimesteps=ntimesteps)
 
     marginal_damages = -1 * mm[:neteconomy, :C][1:ntimesteps, :] * 10.0^12 * 12/44 # convert from trillion $/ton C to $/ton CO2; multiply by -1 to get positive value for damages
-    global_marginal_damages = dropdims(sum(marginal_damages, dims = 2), dims=2)
 
-    global_c = dropdims(sum(mm.base[:neteconomy, :C], dims = 2), dims=2)
-    global_pop = dropdims(sum(mm.base[:neteconomy, :l], dims = 2), dims=2)
-    global_cpc = 1000 .* global_c ./ global_pop
+    # get consumption and population
+    c = mm.base[:neteconomy, :C]
+    pop = mm.base[:neteconomy, :l]
+    
+    num_regions = length(dim_keys(mm.base, :regions))
+
+    # get consumption per capita matrix
+    if equity_weighting
+        cpc = 1000 .* (c ./ pop)
+    else
+        cpc = 1000 .* repeat(sum(c, dims = 2) ./ sum(pop, dims = 2), 1, num_regions)
+    end
 
     year_index = findfirst(isequal(year), model_years)
+    
+    if isnothing(normalization_region)
```

Comment:
Do we need this calculation? Couldn't we just index into the already calculated `cpc` terms from above using the `year_index` for the specified normalization region?

Reply:
@FrankErrickson you're right, we can probably simplify here by using the `cpc` from above, with something like the following.
`cpc_0 = isnothing(normalization_region) ?  mean(cpc[year_index, :]) : cpc[year_index, normalization_region]
`